Close #96 - one variable to remember sign in (NSUserDefaults)

tiffanyfong opened the pull request:
NSUserDefaults is specific to device + faster than Firebase

tiffanyfong committed:
Close #96 - one variable to remember sign in

daking2014 reviewed (commented):
So the way I am doing notifications right now is that when the user signs in, I check and make sure that there is one morning notification scheduled for each of the next 7 days. Any ideas how to replicate this if the user is remembered and thus doesn't actually sign in?

tiffanyfong committed:
#97 make 'abstract' Question class - in progress

tiffanyfong committed:
i forgot what this is

tiffanyfong commented:
@hmc-cs-asheppard @hmc-cs-dlan @hmc-cs-daking 

Lots of things changed. BUT NOW REMEMBER ME WORKS 
1. Added an "external dependency" for keychain for secure data storage. A few objective-c files have been added. I used a keychain to store the uid (user id) and then let firebase retrieve the rest of the info.
https://www.raywenderlich.com/92667/securing-ios-data-keychain-touch-id-1password
1. AppState.sharedInstance.signedIn is never used. I am using NSUserDefaults (for bool signedIn) because this is persistent even when I close and reopen the app. AppState resets when you close the app.
2. Instead of "name" or "email", I used "userName" and "userEmail" for consistency with Firebase
3. Added a "showLogin" function in LoginVC that can be accessed by other files.
4. I attempted to make an "abstract" Question class, but it's unused and not ready.

tiffanyfong commented:
Close #134

tiffanyfong committed:
remember me works with keychain
